src/github/createPRViewProvider.ts

```diff
@@ -188,9 +188,11 @@ export class CreatePullRequestViewProvider extends WebviewViewBase implements vs
 		let branchesForCompare = branchesForRemote;
 		if (defaultCompareRemote.owner !== defaultBaseRemote.owner) {
 			branchesForCompare = await this._folderRepositoryManager.listBranches(defaultCompareRemote.owner, defaultCompareRemote.repositoryName);
-			if (!branchesForCompare.includes(this.compareBranch.name!)) {
-				branchesForCompare.push(this.compareBranch.name!);
-			}
+		}
+
+		if (!branchesForCompare.includes(this.compareBranch.name!)) {
```

Comment:
nit: is is worth having a check for the name to avoid the `!`? (here and in other places)

Reply:
yeah, good point! added some additional checks for this